manifests/container.pp

```diff
@@ -0,0 +1,46 @@
+# @summary Pulp Container plugin
+# @param location_prefix
+#   In the Apache configuration a location with this prefix is exposed. The
+#   version will be appended.
+# @param registry_v1_path
+#   The path beneath the location prefix to forward. This is also appended to
+#   the content base url.
+# @param registry_v2_path
+#   The path beneath the location prefix to forward. This is also appended to
+#   the content base url.
+class foreman_proxy_content::container (
+  String $location_prefix = '/pulpcore_registry',
```

Comment:
This path overlaps with https://github.com/theforeman/puppet-pulpcore/blob/e3e27dd380823ac696019ccb0f6cc64aefba99d5/manifests/plugin/container.pp#L9. It feels at least confusing but probably unreliable to have two reverse proxy configs for the same path. I think it's better to make `pulpcore::plugin::container` configurable so it accepts a different authentication method.

For what it's worth, I thought about doing the exact same thing for **all** pulpcore authentication to remove the need for a specific client certificate from the Katello side. Looks like I haven't filed an issue for it, but it's roughly the same as https://projects.theforeman.org/issues/30368. Foreman has a client certificate to connect to Smart Proxies. We can use the same client certificates for Pulp 3.

@jlsherrill @ehelms I think we've talked about this but haven't scheduled anything. It would greatly simplify a lot of things that we need to do regarding certificates.

Reply:
I suppose it would be better for all the directives to be under the same `<Location "/pulpcore_registry/v2/">`, but I believe the two `Location`s with the same path will just be merged (http://httpd.apache.org/docs/current/sections.html#merging).  I tried reversing the order and it doesn't seem to matter either.



> For what it's worth, I thought about doing the exact same thing for **all** pulpcore authentication to remove the need for a specific client certificate from the Katello side. Looks like I haven't filed an issue for it, but it's roughly the same as https://projects.theforeman.org/issues/30368. Foreman has a client certificate to connect to Smart Proxies. We can use the same client certificates for Pulp 3.
> 
> @jlsherrill @ehelms I think we've talked about this but haven't scheduled anything. It would greatly simplify a lot of things that we need to do regarding certificates.

Perhaps a goal for 4.1? Just want to make sure we're not late for 4.0.